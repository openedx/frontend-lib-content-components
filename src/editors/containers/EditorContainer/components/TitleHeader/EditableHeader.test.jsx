import React from 'react';
import { shallow } from '@edx/react-unit-test-utils';
import { Form } from '@edx/paragon';
import * as module from './EditableHeader';
import EditConfirmationButtons from './EditConfirmationButtons';

describe('EditableHeader', () => {
  const props = {
    handleChange: jest.fn().mockName('args.handleChange'),
    updateTitle: jest.fn().mockName('args.updateTitle'),
    handleKeyDown: jest.fn().mockName('args.handleKeyDown'),
    inputRef: jest.fn().mockName('args.inputRef'),
    localTitle: 'test-title-text',
    cancelEdit: jest.fn().mockName('args.cancelEdit'),
  };
  let el;
  beforeEach(() => {
    el = shallow(<module.EditableHeader {...props} />);
  });

  describe('snapshot', () => {
    test('snapshot', () => {
      expect(el.snapshot).toMatchSnapshot();
    });
    test('displays Edit Icon', () => {
      const formControl = el.instance.findByType(Form.Control)[0];
      expect(formControl.props.trailingElement).toMatchObject(
        <EditConfirmationButtons updateTitle={props.updateTitle} cancelEdit={props.cancelEdit} />,
      );
    });
  });
});
