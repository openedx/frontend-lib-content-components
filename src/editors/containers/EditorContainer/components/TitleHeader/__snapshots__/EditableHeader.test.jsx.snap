// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditableHeader snapshot snapshot 1`] = `
<Form.Group
  onBlur={[Function]}
>
  <Form.Control
    autoFocus={true}
    onChange={[MockFunction args.handleChange]}
    onKeyDown={[MockFunction args.handleKeyDown]}
    placeholder="Title"
    style={
      {
        "paddingInlineEnd": "calc(1rem + 84px)",
      }
    }
    trailingElement={
      <injectIntl(ShimmedIntlComponent)
        cancelEdit={[MockFunction args.cancelEdit]}
        updateTitle={[MockFunction args.updateTitle]}
      />
    }
    value="test-title-text"
  />
</Form.Group>
`;
